lib/mdl/ruleset.rb

```diff
@@ -31,6 +33,17 @@ def params(params = nil)
       @params.update(params) unless params.nil?
       @params
     end
+
+    def docs(url)
+      raise 'A docs url is already set within this rule' if @docs_overridden
```

Comment:
When might we hit this condition? We already support `docs` being called for both the ruleset and the rule, is this like the case where someone does:
```
rule 'LOCAL01', 'my cool rule' do
  docs {... }
  check { ... }
  docs { ... }
end
```
?

Also, thinking outloud here.. should we support a block on the rule-level docs DSL method? I guess we have all the info we need visually because we're writing it into a specific rule, but making the user type it again seems odd, so maybe accepting a block is useful?

Reply:
Yeah, that's the case I was protecting against — it might be really confusing if you're calling docs twice and wondering why it's not what it appears to be. (It's pretty unlikely!)

Accepting a block inside `Rule#docs` seemed unnecessary, as the only cases I could think of where you'd have the ID or name in the URL is if you did that for _multiple_ rules (so you'd have it on the `RuleSet` instead) — I'm not against it if you can see value there!

Reply:
I'm thinking of the case where like you have a one-off:

```ruby
docs { |id, name| "https://example.com/#{id}" }
rule {...}
rule {...}
rule {
  docs { |id, name|, https://othersite.com/#{id}" }
}
rule {...}
```

Obviously they could just put that the end with a different rulset-docs block, or they could put an `if` in the ruleset-doc block to check for ID, or whatever, but I figure ... it's trivial to make the above easy, so why not?

Reply:
Did you see this comment here?

Reply:
My apologies! I've had an apparently similarly busy year — I hope the change I've just submitted addresses just this — it adds the ability to specify dynamically generated docs links within rules (as well as within rulesets).

I had some difficulty getting my (new instruction set) laptop running the tests, so I patched blind, but I'll work though any spec failures I see when the CI is complete.

Thank you for your pateience!